### PR TITLE
Add versions to deprecated warnings

### LINKED ISSUE #4261: [FEATURE] Add version deprecated information to PTB raised warnings
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

We can slightly increase the information provided in warnings by adding which version the feature was deprecated in.

### Describe the solution you'd like

[Python's `warn()` ](https://docs.python.org/3.13/library/warnings.html#warnings.warn)accepts instances of `UserWarning`, so we can customize the message without repeating the phrase "Deprecated in NEXT.VERSION" everywhere.

Steps:

1. Add an argument `version` and `message` to `PTBDeprecationWarning`.
2. Override `__str__` to add the phrase `"Depreacted in {self.version}: {self.message}"`

Our call will now look like: 
```python
PTBDeprecationWarning("NEXT.VERSION", "msg")
```
### Describe alternatives you've considered

Simply add the version information in the warning message, but then we'll have to keep repeating ourselves.

### Additional context

We can also do the same for `PTBUserWarning` and `PTBRuntimeWarning`.

### PR DESCRIPTION
**Pull request by harshil21**
Closes #4261. To be merged after #4243

### CODE REVIEW

File: telegram/_chat.py [hunk 1]
```diff
@@ -910,10 +910,12 @@ def __init__(
             for arg in _deprecated_attrs:
                 if (val := object.__getattribute__(self, arg)) is not None and val != ():
                     warn(
-                        f"The argument `{arg}` is deprecated and will only be available via "
-                        "`ChatFullInfo` in the future.",
+                        PTBDeprecationWarning(
+                            "NEXT.VERSION",
```
Comment (Bibo-Joshi) [inline telegram/_chat.py:914]: Just as reminder: In cas this doesn't make it into the same release as #4260, this needs updating

File: telegram/_chatfullinfo.py [hunk 2]
```diff
@@ -161,6 +161,6 @@ def __init__(
 
         # Required and unique to this class-
         with self._unfrozen():
-            self.max_reaction_count = max_reaction_count
+            self.max_reaction_count: int = max_reaction_count
```
Comment (Bibo-Joshi) [inline telegram/_chatfullinfo.py:164]: This looks like it should rather go into #4260

File: telegram/warnings.py [hunk 3]
```diff
@@ -56,4 +56,11 @@ class PTBDeprecationWarning(PTBUserWarning, DeprecationWarning):
...
 
-    __slots__ = ()
+    __slots__ = ("message", "version")
+
+    def __init__(self, version: str, message: str) -> None:
+        self.version: str = version
+        self.message: str = message
+
+    def __str__(self) -> str:
+        return f"Deprecated since version {self.version}: {self.message}"
```
Comment (Bibo-Joshi) [inline telegram/warnings.py:66]: Since this module is public, the changes should be documented in the docstrings

File: tests/test_bot.py [hunk 4]
```diff
@@ -2076,6 +2076,7 @@ async def make_assertion(url, request_data: RequestData, *args, **kwargs):
         monkeypatch.setattr(bot.request, "post", make_assertion)
         assert await bot.do_api_request("camel_case")
 
+    @pytest.mark.filterwarnings("ignore::telegram.warnings.PTBUserWarning")
```
Comment (Bibo-Joshi) [inline tests/test_bot.py:2079]: why is this needed now when it wasn't needed before? Some for the other places

File: tests/test_bot.py [hunk 5]
```diff
@@ -4061,7 +4063,7 @@ async def test_set_message_reaction(self, bot, chat_id, message):
 
     @pytest.mark.parametrize("bot_class", [Bot, ExtBot])
     async def test_do_api_request_warning_known_method(self, bot, bot_class):
-        with pytest.warns(PTBDeprecationWarning, match="Please use 'Bot.get_me'") as record:
+        with pytest.warns(PTBUserWarning, match="Please use 'Bot.get_me'") as record:
```
Comment (Bibo-Joshi) [inline tests/test_bot.py:4066]: Why is the change necessary?

**Review by Bibo-Joshi — COMMENTED** [review]
Nice idea, I like it :) Left some questions though

File: tests/test_bot.py (hunk 4 above)
Comment (harshil21) [reply tests/test_bot.py:2079]: I changed the warning of `Bot.do_api_request` from a DeprecationWarning to UserWarning since DeprecationWarning there didn't make sense, nothing was being deprecated there

File: tests/test_bot.py (hunk 5 above)
Comment (harshil21) [reply tests/test_bot.py:4066]: same reason above

File: tests/test_bot.py (hunk 4 above)
Comment (Bibo-Joshi) [reply tests/test_bot.py:2079]: Ah, now I get it 👍 I used deprecationwarning there b/c it made sense to me to say "Using `do_api_request` for endpoint XY is now deprecated, since we have a dedicated method for that". But I'm not married to that

**Review by Bibo-Joshi — APPROVED** [review]
